### PR TITLE
Changing config file to initialize for storage

### PR DESCRIPTION
**Pull request by SiobhanPowell**

### CODE REVIEW

File: network_generation.py [hunk 1]
```diff
@@ -39,4 +41,20 @@ def six_bus(vn_high=20, vn_low=0.4, length_km=0.03, std_type='NAYY 4x50 SE'):
                    name='line5')
     pp.create_line(net, 5, 7, length_km=length_km, std_type=std_type,
                    name='line6')
+    #  add controllable storage
+    num_batteries = np.shape(battery_locations)[0]
+    for i in range(num_batteries):
+        pp.create_storage(net, bus=battery_locations[i], p_kw=0.0, max_e_kwh=energy_capacity, soc_percent=init_soc)
```
Comment (bmeyers) [inline network_generation.py]: Let's align this with `NetModel.add_battery`. There are additional fields that need to be included, as well as non-standard fields.

**Review by bmeyers — CHANGES_REQUESTED** [review]

**Review by bmeyers — APPROVED** [review]

File: pp_network.py [hunk 2]
```diff
@@ -17,12 +17,15 @@ class NetModel(object):
...
     def __init__(self, net_given=None, network_name='rural_1',
                  zero_out_gen_shunt_storage=False, tstep=1./60,
-                 net_zero_reward=1.0, env_name=None, baseline=True):
+                 net_zero_reward=1.0, env_name=None, baseline=True, config=None):
         """Initialize attributes of the object and zero out certain components
         in the standard test network."""
 
         if env_name is not None:
-            self.config = get_config(env_name, baseline)
+            if config is not None:
```
Comment (bmeyers) [inline pp_network.py]: it's redundant to have the user give both an environment name and a config file. If the user wants to pass a custom config file for which there isn't a name assigned yet, this code won't let them do that. So, let's change the logic to check if a config object is passed first (if so, use it), and if there isn't one, then try to load one using `get_config(env_name, baseline)`.

**Review by bmeyers — CHANGES_REQUESTED** [review]

File: network_generation.py [hunk 3]
```diff
@@ -1,13 +1,34 @@
...
+def add_battery(net, bus_number, p_init, energy_capacity, init_soc=0.5,
+                max_p=50, min_p=-50, eff=1.0, capital_cost=0, min_e=0.):
+    """Change the network by adding a battery / storage unit. """
+    pp.create_storage(net, bus_number, p_init, energy_capacity,
+                      soc_percent=init_soc, max_p_kw=max_p, min_p_kw=min_p,
+                      min_e_kwh=min_e)
+    # if 'eff' not in net.storage.columns:
+    #     net.storage['eff'] = eff
+    # else:
+    idx = net.storage.index[-1]
```
Comment (bmeyers) [inline network_generation.py:18]: Were you able to test that this worked correctly?

File: pp_network.py (hunk 2 above)
Comment (SiobhanPowell) [reply pp_network.py]: Changed it, good catch!

File: network_generation.py (hunk 3 above)
Comment (SiobhanPowell) [reply network_generation.py:18]: Yes it tests okay, I'll remove the commented out older parts

**Comment by SiobhanPowell** [comment]
I think I made all the changes and it should be good to merge! Let me know if you see any other issues. I will start another branch to rearrange the config and NetModel pieces as discussed

**Review by bmeyers — APPROVED** [review]

**Comment by bmeyers** [comment]
Looks great, merging!

**Pull request merged**